libmongocxx_ros/CMakeLists.txt

```diff
@@ -1,250 +1,81 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(libmongocxx_ros)
 
-
-
-## Compile as C++11, supported in ROS Kinetic and newer
-# add_compile_options(-std=c++11)
-
-## Find catkin macros and libraries
-## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
-## is used, also find other catkin packages
-find_package(catkin REQUIRED COMPONENTS
-)
-
-## System dependencies are found with CMake's conventions
-# find_package(Boost REQUIRED COMPONENTS system)
-
-include(ExternalProject)
-
+find_package(catkin REQUIRED)
+
+# find MongoClient from system first
+# if not found, downloads and compiles from sources
+set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/" ${CMAKE_MODULE_PATH})
+find_package(MongoClient)
+if (MongoClient_FOUND)
+  message(STATUS "Found System MongoClient. Copying them into ROS workspace")
+  add_custom_command(OUTPUT ${CATKIN_DEVEL_PREFIX}/include
+    COMMAND ${CMAKE_COMMAND} -E make_directory ${CATKIN_DEVEL_PREFIX}/include)
+  add_custom_command(OUTPUT ${CATKIN_DEVEL_PREFIX}/include/mongo
+    COMMAND ${CMAKE_COMMAND} -E copy_directory ${MongoClient_INCLUDE_DIR}/mongo ${CATKIN_DEVEL_PREFIX}/include/mongo
+    DEPENDS ${CATKIN_DEVEL_PREFIX}/include)
+  add_custom_command(OUTPUT ${PROJECT_SOURCE_DIR}/include/mongo
+    COMMAND ${CMAKE_COMMAND} -E copy_directory ${CATKIN_DEVEL_PREFIX}/include/mongo ${PROJECT_SOURCE_DIR}/include/mongo
+    DEPENDS ${CATKIN_DEVEL_PREFIX}/include/mongo)
+  add_custom_command(OUTPUT ${CATKIN_DEVEL_PREFIX}/lib
+    COMMAND ${CMAKE_COMMAND} -E make_directory ${CATKIN_DEVEL_PREFIX}/lib)
+  add_custom_command(OUTPUT ${CATKIN_DEVEL_PREFIX}/lib/libmongoclient.so
+    COMMAND ${CMAKE_COMMAND} -E copy ${MongoClient_LIBRARY} ${CATKIN_DEVEL_PREFIX}/lib/libmongoclient.so
+    DEPENDS ${CATKIN_DEVEL_PREFIX}/lib)
+  add_custom_target(mongocxx
+    DEPENDS
+    ${CATKIN_DEVEL_PREFIX}/lib/libmongoclient.so
+    ${PROJECT_SOURCE_DIR}/include/mongo
+    ${CATKIN_DEVEL_PREFIX}/include/mongo)
+else()
+  message(STATUS "Not Found System MongoClient. Downloading...")
```

Comment:
are you assuming this is kinetic situation?

Reply:
@k-okada Yes. In indigo (=ubuntu14.04) system mongoclient library is used, in kinetic (=ubuntu16.04) mongoclient is downloaded and compiled from source.